data/input_files/tests/meshing/meshing.lua

```diff
@@ -0,0 +1,22 @@
+main_mesh_from_file = {
+    -- mesh file
+    mesh = "../../../meshes/beam-hex.mesh",
+    -- serial and parallel refinement levels
+    ser_ref_levels = 1,
+    par_ref_levels = 0,
+}
+
+main_mesh_cuboid = {
+    -- mesh file
+    elements = {x = 3, y = 3, z = 3},
+    size = {x = 1, y = 2, z = 3},
+    ser_ref_levels = 1,
+    par_ref_levels = 0,
+}
+
+main_mesh_rect = {
+    -- mesh file
+    elements = {x = 3, y = 3},
+    ser_ref_levels = 1,
+    par_ref_levels = 0,
+}
```

Comment:
In the test input files, we could also look at using the approach we use elsewhere in our input files for variants (e.g., for solver options):
```lua
main_mesh = {
  type = "from_file",
  from_file_options = {
    file = "/path/to/mesh"
  },
  generated_options = {
    shape = "cuboid",
    elements = {x = 3, y = 3, z = 3},
    size = {x = 1, y = 2, z = 3},
  }
  ser_ref_levels = 1,
  par_ref_levels = 0,
}
```

Reply:
I see. Now that I think about it, it's better to have a `type` parameter because it forces the user to choose either `file` or `generated`. Otherwise it might be unclear which is active.